### ISSUE TITLE
Problem Statement Attachments Are Not Deleted On Exercise Deletion

### ISSUE THREAD

**Issue by b-fein** [ls1intum/Artemis#4866]
### Describe the bug

It is possible to upload images that are then displayed in the problem statement. Those images are saved next to the Artemis war in the filesystem in the folder `uploads/markdown/`.

When deleting the exercise, the no longer needed files should be deleted as well to not fill up the host file system unnecessarily.

### To Reproduce

1. Create an exercise.
2. Upload some files into the problem statement. Those files are saved in the folder mentioned above on the server.
3. Delete the exercise.
4. The files are still there.

### Expected behavior

The files should be deleted on exercise deletion.

### What browsers are you seeing the problem on?

Chrome, Safari, Microsoft Edge, Firefox, Other (specify in "Additional context")

### Additional context

I think the same applies to drag-and-drop quiz backgrounds as well.

**Comment by just-max** [comment]
Sometimes we use the same image across multiple exercises after only uploading it once. This would cause images to unexpectedly disappear from other those exercises after deleting the first.

**Comment by ge65cer** [comment]
I agree with @just-max here, GitHub for example also does not delete images uploaded to its Markdown editor if you delete the associated comment.
